File: algorithms/cpp/461_hamming_distance/hamming_distance_1.cpp
//============================================================================
// Name        : hamming_distance_1.cpp
// Author      : PY131
// Version     :
// Copyright   : 
// Description : C++, Ansi-style
//============================================================================

#include <iostream>
using namespace std;

class Solution {
public:
    int hammingDistance(int x, int y) {
        int ham_dis = 0;
        while(x!=0 || y!=0){
            if(x%2 != y%2) ham_dis++;
            x = x>>1;
            y = y>>1;
        }
        return ham_dis;
    }
};

int main() {
	Solution hd1;
	int x1, y1;
	cout << "please input two integers: " << endl;
	cin >> x1 >> y1;
	cout << "hamming distance is: ";
	cout << hd1.hammingDistance(x1, y1) << endl;
}

File: algorithms/java/461_hamming_distance/hamming_distance_1.java
import java.util.Scanner;

public class hamming_distance {
    
	public int hammingDistance(int x, int y) {
        int ham_dis;
        int z = x ^ y; 
        for(ham_dis=0; z!=0; z=z>>1){
        	if(z%2 != 0) ham_dis++;
        }
        return ham_dis;
    }
	
	public static void main(String[] args) {
		int x1, x2;
		hamming_distance hd1 = new hamming_distance();
		Scanner sc1 = new Scanner(System.in);
		
		System.out.println("input two integers:");
		x1 = sc1.nextInt();
		x2 = sc1.nextInt();
		System.out.printf("hamming distance is: %d", hd1.hammingDistance(x1, x2));
		sc1.close();
	}

}
